src/service/file-service.js

```diff
@@ -45,26 +44,34 @@ class FileService {
         return this._readFile(filePath, true);
     }
 
+    async fileExists(filePath) {
+        try {
+            await stat(filePath)
```

Comment:
Use exists method or something like that. Don't use catch to handle normal code execution.

Reply:
this is the only way I found to check if file exists without blocking the event loop. Should I do it synchronously ?

Reply:
Yeah, i see that now, ok than keep this implementation